core/src/main/java/org/elasticsearch/index/query/functionscore/DecayFunctionParser.java

```diff
@@ -66,24 +61,19 @@
  * parameters origin and scale.
  * <p>
  * To write a new scoring function, create a new class that inherits from this
- * one and implements {@link #getBuilderPrototype()} and {@link #getNames()}.
- * Also create its corresponding {@link DecayFunctionBuilder}. The latter needs to
- * implement {@link DecayFunctionBuilder#doReadFrom(StreamInput)} and
- * {@link DecayFunctionBuilder#doWriteTo(StreamOutput)} for serialization purposes,
- * {@link DecayFunctionBuilder#doEquals(DecayFunctionBuilder)} and
- * {@link DecayFunctionBuilder#doHashCode()} for equality checks,
- * {@link DecayFunctionBuilder#getName()} that returns the name of the function and
- * {@link DecayFunctionBuilder#getDecayFunction()} which returns the corresponding lucene function.
- * <p>
- * See {@link GaussDecayFunctionBuilder} and {@link GaussDecayFunctionParser}
- * for an example. The parser furthermore needs to be registered in the
- * {@link org.elasticsearch.search.SearchModule SearchModule}.
- *
+ * one and create a corresponding {@link DecayFunctionBuilder} and register it
+ * using
+ * {@link org.elasticsearch.search.SearchModule#registerScoreFunction(ParseField, ScoreFunctionParser, String, Writeable.Reader)}.
+ * See {@link GaussDecayFunctionBuilder#PARSER} for an example.
  */
-
-public abstract class DecayFunctionParser<DFB extends DecayFunctionBuilder<DFB>> implements ScoreFunctionParser<DFB> {
+public class DecayFunctionParser<DFB extends DecayFunctionBuilder<DFB>> implements ScoreFunctionParser<DFB> {
 
     public static final ParseField MULTI_VALUE_MODE = new ParseField("multi_value_mode");
+    private final BiFunction<String, BytesReference, DFB> createFromBytes;
+
+    public DecayFunctionParser(BiFunction<String, BytesReference, DFB> createFromBytes) {
```

Comment:
This is great, but It took me a bit to understand why we need it and how it is used. Maybe this can be explained in a short comment?